app/src/main/java/com/sopt/dive/ui/screens/LoginScreen.kt

```diff
@@ -0,0 +1,91 @@
+package com.sopt.dive.ui.screens
+
+import android.widget.Toast
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sopt.dive.viewmodel.MainViewModel
+import com.sopt.dive.ui.components.CommonButton
+import com.sopt.dive.ui.components.CommonInputField
+
+
+@Composable
+fun LoginScreen(
+    mainViewModel: MainViewModel,
+    onLoginSuccess: (String, String) -> Unit,
+    onSignUpClick: () -> Unit
+) {
+    val context = LocalContext.current
+
+    var idText by remember { mutableStateOf("") }
+    var pwText by remember { mutableStateOf("") }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(20.dp)
+    ) {
+        Text(
+            text = "Welcome To SOPT",
+            modifier = Modifier.fillMaxWidth(),
+            textAlign = TextAlign.Center,
+            fontWeight = FontWeight.Bold,
+            fontSize = 30.sp
+        )
+
+        CommonInputField(
+            titleText = "id",
+            value = idText,
+            onValueChange = { idText = it },
+            placeMessage = "아이디를 입력해주세요",
+            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next, keyboardType = KeyboardType.Text)
+        )
+
+        CommonInputField(
+            titleText = "pw",
+            value = pwText,
+            onValueChange = { pwText = it },
+            placeMessage = "비밀번호를 입력해주세요",
+            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Done, keyboardType = KeyboardType.Password),
+            visualTransformation = PasswordVisualTransformation()
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        CommonButton(
+            onClick = {
+                if (mainViewModel.loginUser(idText, pwText)) {
+                    onLoginSuccess(idText, pwText)
+                } else {
+                    Toast.makeText(context, "로그인 실패", Toast.LENGTH_SHORT).show()
+                }
+            },
+            textMessage = "로그인"
+        )
+
+        Text(
+            text = "회원가입하기",
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(bottom = 40.dp)
+                .clickable { onSignUpClick() },
```

Comment:
불필요한 람다 객체 생성을 줄이기 위해서 이렇게 함수 자체를 넘겨주시는거 어떠세요??

```suggestion
                .clickable (onClick = onSignUpClick),
```

Reply:
감사합니다 ! 소희님의 코드 리뷰 이후, 불필요한 람다 객체 생성에 대해 조금 더 공부하면서, 공부한 내용을 정리해보니 , 제가 이해한게 맞는지 확인 부탁드려도 될까요 ?? 

 `하나의 역할을 실행한다면 함수로 넘겨줄 수 있기에, 람다생성보다 함수 자체가 적합하고 두 개 이상의 역할이라면 람다식으로 넘겨주어야한다` 라고 이해했는데 제대로 이해한게 맞을까요 ? 🤔

Reply:
감사합니다! 소희님의 코드 리뷰 이후로 불필요한 람다 객체 생성에 대해 조금 더 공부해봤습니다.
공부한 내용을 정리해보니, _하나의 역할만 수행한다면 함수 참조를 직접 전달하는 것이 더 적합하고,
두 개 이상의 동작을 함께 수행해야 한다면 람다식으로 작성하는 것이 적합하다_ 라고 이해했습니다.
혹시 이 이해가 맞을까요? 🤔